lasy/utils/laser_utils.py

```diff
@@ -459,3 +467,183 @@ def create_grid(array, axes, dim):
         assert grid.field.shape == array[np.newaxis].shape
         grid.field = array[np.newaxis]
     return grid
+
+
+def export_to_z(dim, grid, omega0, z_axis=None, z0=0.0, t0=0.0, backend="NP"):
+    """
+    Export laser pulse to spatial domain from temporal domain (internal LASY representation).
+
+    Parameters
+    ----------
+    dim : string
+        Dimensionality of the array. Options are:
+        - 'xyt': The laser pulse is represented on a 3D grid:
+                 Cartesian (x,y) transversely, and temporal (t) longitudinally.
+        - 'rt' : The laser pulse is represented on a 2D grid:
+                 Cylindrical (r) transversely, and temporal (t) longitudinally.
+
+    grid : a Grid object.
+        It contains a ndarrays (V/m) with
+        the value of the envelope field and the associated metadata
+        that defines the points at which the laser is defined.
+
+    omega0 : scalar
+        Angular frequency at which the envelope is defined.
+
+    z_axis : 1D ndarray of doubles (optional)
+        Spatial `z` axis along which the field should be reconstructed.
+        If not provided, `z_axis = c * t_axis` is considered.
+
+    z0 : scalar (optional)
+        Position from which the field is produced (emitted).
+
+    t0 : scalar (optional)
+        Moment of time at which the field is produced.
+
+    backend : string (optional)
```

Comment:
could we mention here a list of the possible values?

Reply:
I am not sure we want to support this at the moment, so I just redirected to Axiprop in a slightly more specific way than was before.